onmt/inference.py

```diff
@@ -15,19 +17,96 @@
 
 model = Img2LatexModel(name=model_path, opt=opt)
 
-def predict_file(file_name):
+def _make_batches_from_samples(samples, batch_size):
+    """
+    :param samples:
+    :param batch_size:
+    :return: batches, each batches contains at most "batch_size" samples
+    """
+    samples_len = len(samples)
+    batches = [(start, min(start + batch_size, samples_len)) for start in range(0, samples_len, batch_size)]
+
+    return batches
+
+def _predict_with_batch(file_names, batch_size):
+    """
+
+    :param file_names: list of string (image file name in this case)
+    :param batch_size: number of samples to be processed
+    :return: predictions, confident scores
+    """
+    assert batch_size >= 1
+
+    batches = _make_batches_from_samples(samples=file_names, batch_size=batch_size)
+
+    for (start, end) in batches:
+        filename_inputs = file_names[start:end]
+        try:
+            inputs = [cv2.imread(str(filename_input), 0) for filename_input in filename_inputs]
+
+            predictions, confident_scores = model.predict(inputs, batch_size=batch_size)
+
+            yield (predictions, confident_scores)
+        except:
+            warning_message = "Error occurred when processing the below collection of file names: \n %s \n" \
+                              ". Try to use batch size = 1 instead." % str(filename_inputs)
+            logging.warning(warning_message)
+
+            predictions, confident_scores = [], []
+
+            for filename_input in filename_inputs:
+                try:
+                    input = [cv2.imread(str(filename_input))]
+                    prediction, confident_score = model.predict(input, batch_size=1)
+
+                    predictions += prediction
+                    confident_scores += confident_score
+                except:
+                    warning_message = "Error occurred when processing the file name: %s. Skip this file name. " % \
+                                      filename_input
+                    logging.warning(warning_message)
+
+                    predictions += [None]
+                    confident_scores += [None]
+
+            yield (predictions, confident_scores)
+
+
+def predict_file(file_name, batch_size=1):
     global model
 
-    results = {}
     try:
-        predict, confident = model.predict([cv2.imread(file_name, 0)])
+        file_names, results = [], []
+        if type(file_name) is str or type(file_name) is Path:
```

Comment:
Path(file_name) will create a subclass of Path depending on the OS. In my case it was PosixPath so type(file_name) == Path will be False
```suggestion
        if type(file_name) is str or isinstance(file_name, Path):
```